packages/gestalt/src/Button.js

```diff
@@ -72,19 +95,22 @@ export default function Button(props: Props) {
       onClick={event => onClick && onClick({ event })}
       type={type}
     >
-      <Text
-        align="center"
-        color={
-          (disabled && 'gray') ||
-          (selected && 'white') ||
-          textColor ||
-          DEFAULT_TEXT_COLORS[color]
-        }
-        overflow="normal"
-        weight="bold"
-      >
-        {text}
-      </Text>
+      {iconEnd ? (
```

Comment:
@dangerismycat could you also add a simple (snapshot) test to verify the icon shows up when we add `iconEnd`?